### ISSUE TITLE
Time for a new release (1.98) ?

### ISSUE THREAD

**Issue by rtissera** [irixxxx/picodrive#14]
I can see a lot of great work and bugfixed has been done between latest 1.97 release and git HEAD, have you plan to make a 1.98 release soon ?

**Comment by irixxxx** [comment]
Not necessarily... originally I wanted to add chd support in 1.98... I'll think about it.

**Comment by rtissera** [comment]
chd integration would be great !
well as I try to maintain http://github.com/rtissera/libchdr/ let me know if you have questions or issues integrating it.

**Comment by irixxxx** [comment]
Yes, there are problems... the current handling of input media isn't fitting in well with the libchdr api. This need some revising and restructuring before this can happen.

**Comment by irixxxx** [comment]
Is there something in master you would need to have in a release? I could always do something like 1.97a, or 1.98-alpha1... somewhat ugly, but who cares.

**Comment by rtissera** [comment]
> Is there something in master you would need to have in a release? I could always do something like 1.97a, or 1.98-alpha1... somewhat ugly, but who cares.

Nothing which cannot wait. CHD integration is a big milestone enough to wait a new proper release.

**Comment by 0x90shell** [comment]
CHD support would be huge. 👍

**Comment by irixxxx** [comment]
I'm making progress on CHD. I expect some news about this within the next 2 weeks.

**Comment by rtissera** [comment]
Great news let me know I will be happy to integrate in upcoming Batocera version !

If you have any feedback request for libchdr I am here to help.

Envoyé de mon iPhone

**Comment by irixxxx** [comment]
I have it in mostly working condition on x86. ATM I'm looking at how to integrate this into picodrive's build environment. Now, I have encountered some problems:
- cmake complains about TRANSFORM and JOIN not available?
- same for target_link_options?
- I can't link against libchdr-static since it contains lto objects
- how can I cross-compile libchdr?

**Comment by irixxxx** [comment]
I have decided to push preliminary chd support for reviewing. Feel free to comment.

**Comment by rtissera** [comment]
Amazing I will try to see if I can help for your cross compiling
environment and Cmake issues

**Comment by irixxxx** [comment]
I've already found out the cmake in my development environment is too old. It's version 3.5, whilst the 1st two items on my list require 3.13. If I don't want to put some effort into compiling my own arm toolchain I'm kind of locked into that environment since I can't use anything newer than gcc 4.7 for gp2x compiling :-D
I could check for libchdr in configure and only use it if it's available... I have however the impression it's probably not usually there.

**Comment by rtissera** [comment]
I'm afraid chd decompression on GP2X hardware will be a CPU hog especially for CHDv5 compressed files containing LZMA chunks

**Comment by irixxxx** [comment]
I'm pretty much sure no one will ever use it on that. GPH hardware should use cue/iso with mp3/wav, if anything.

**Comment by irixxxx** [comment]
Hmm. I currently see the inability to link chdr statically as the main problem. I'm normally releasing binaries which are self-contained but for SDK/system libraries.
It has those LTO objects which I can't link to, and the depency stuff is missing alltogether in libchdr-static.

**Comment by irixxxx** [comment]
IMO cross compiling is a bit weird. With this I get it to use the right compiler:
`cmake -DCMAKE_C_COMPILER=/usr/bin/arm-linux-gnueabi-gcc`
`make`
But this doesn't seem to be the recommended way to do it, internet says to use a toolchain file.
Besides, how can I hand down additional arguments like `--sysroot`, or additional include and library paths, down to the compiler? Or is it really necessary to auto-generate a toolchain-file somehow out of my make environment?

BTW, one of my compiler complains about an error in `src/libchdr_chd.c`line 433. That "int scan = 0" there should probably only be "scan = 0".

**Comment by irixxxx** [comment]
Using the PSVita SDK I get undefined references for `fseeko64`and `ftello64`.

**Comment by irixxxx** [comment]
I've integrated libchdr (ATM for CD games only), so release preparations for 1.98 are on the way.

**Comment by rtissera** [comment]
Great news, on my side I merged your dr_flac changes to libchdr, which is imho a great improvement for both developers and end users. Nice example of collaborative effort :) I am looking forward 1.98 and will be happy to integrate it for Batocera.Linux 31

**Comment by irixxxx** [comment]
So... you might have already noticed. I dared call it 1.98. You might give it a go.

**Comment by rtissera** [comment]
amazing great work again :)
